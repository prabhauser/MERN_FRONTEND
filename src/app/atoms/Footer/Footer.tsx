import Breadcrumbs from '@mui/material/Breadcrumbs';
import useStyles from './style';
import { withRouter } from 'react-router-dom';
import { paths } from '../../constants/paths';

const Footer = () => {
    const classes = useStyles();
    return (
        <div>
            {location.hash == `#${paths.LOGIN.path[0]}` || location.hash == `#${paths.LOGIN.path[1]}` ? null : (
                <div className={classes.body}>
                    <Breadcrumbs separator="|" className={classes.Footer}>
                        <span className={classes.textHolder} data-testid="mail">
                            support@dswd.com
                        </span>
                        <span className={classes.textHolder} data-testid="copyRigth">
                            Copyright Â© 2021 DSWD. All rights reserved.
                        </span>
                        <span className={classes.textHolder} data-testid="phoneNumber">
                            (+63)5378697580
                        </span>
                    </Breadcrumbs>
                </div>
            )}
        </div>
    );
};

export default withRouter(Footer);
