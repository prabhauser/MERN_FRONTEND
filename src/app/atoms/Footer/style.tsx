import { makeStyles } from '@material-ui/core';
import { fontSize } from '@mui/system';
import theme from '../../../utilities/theme/theme.module.scss';

export default makeStyles(() => {
    return {
        Footer: {
            height: '47px',
            width: '100vw',
            backgroundColor: '#32355A',
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'center',
            textAlign: 'center',
            color: `${theme.s900bgtxt} !important`
        },

        body: {
            display: 'flex',
            alignItems: 'center',
            marginBottom: '0px',
            marginTop: '20px'
        },
        textHolder: {
            alignItems: 'center !important',
            display: 'flex !important',
            justifyContent: 'center !important',
            height: '15px',
            fontFamily: 'PoppinsSemiBold',
            fontSize: '10px',
            color: `${theme.s900bgtxt} !important`
        }
    };
});
