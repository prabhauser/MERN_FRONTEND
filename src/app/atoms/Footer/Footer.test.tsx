import { shallow } from 'enzyme';
import Footer from './Footer';
const props = {
    mail: '',
    copyRigth: '',
    phoneNumber: ''
};

describe('Footer bar', () => {
    const root = shallow(<Footer />);
    it('test the existence of mail props', () => {
        const wrapper = root.find('[data-testid="mail"]').at(0);
        expect(wrapper).toBeTruthy();
    });
    it('should take Footer snapshot', () => {
        const component = shallow(<Footer />);
        component.update();
        expect(component).toMatchSnapshot();
    });
    it('test the existence of copyRigth props', () => {
        const wrapper = root.find('[data-testid="copyRigth"]').at(0);
        expect(wrapper).toBeTruthy();
    });
    it('test the existence of phoneNumber props', () => {
        const wrapper = root.find('[data-testid="phoneNumber"]').at(0);
        expect(wrapper).toBeTruthy();
    });
});
