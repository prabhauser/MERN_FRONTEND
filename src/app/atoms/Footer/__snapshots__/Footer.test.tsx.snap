// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer bar should take Footer snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
